packages/salesforcedx-vscode-core/src/orgPicker/orgList.ts

```diff
@@ -0,0 +1,192 @@
+
+    if (authFilesArray === null || authFilesArray.length === 0) {
+      return null;
+    }
+    const authInfoObjects: FileInfo[] = [];
+    for (const username of authFilesArray) {
+      try {
+        const filePath = path.join(
+          await AuthInfoConfig.resolveRootFolder(true),
+          '.sfdx',
+          username
+        );
+        const fileData = readFileSync(filePath, 'utf8');
+        authInfoObjects.push(JSON.parse(fileData));
+      } catch (e) {
+        console.log(e);
+      }
+    }
+    return authInfoObjects;
+  }
+
+  public async filterAuthInfo(authInfoObjects: FileInfo[]) {
+    authInfoObjects = authInfoObjects.filter(fileData =>
+      isNullOrUndefined(fileData.scratchAdminUsername)
+    );
+
+    const defaultDevHubUsernameorAlias = await getDefaultDevHubUsernameorAlias();
+    if (defaultDevHubUsernameorAlias) {
+      const defaultDevHubUsername = await OrgAuthInfo.getUsername(
+        defaultDevHubUsernameorAlias
+      );
+      authInfoObjects = authInfoObjects.filter(
+        fileData =>
+          isNullOrUndefined(fileData.devHubUsername) ||
+          (!isNullOrUndefined(fileData.devHubUsername) &&
+            fileData.devHubUsername === defaultDevHubUsername)
+      );
+    }
+
+    const authUsernames = authInfoObjects.map(file => file.username);
+    const aliases = await Aliases.create(Aliases.getDefaultOptions());
+    const authList = [];
+    for (const username of authUsernames) {
+      const alias = await aliases.getKeysByValue(username);
+      if (alias.length > 0) {
+        authList.push(alias + ' - ' + username);
+      } else {
+        authList.push(username);
+      }
+    }
+    return authList;
+  }
+
+  public async updateOrgList() {
+    const authInfoObjects = await this.getAuthInfoObjects();
+    if (isNullOrUndefined(authInfoObjects)) {
+      return null;
+    }
+    const authUsernameList = await this.filterAuthInfo(authInfoObjects);
+    return authUsernameList;
+  }
+}
+
+let statusBarItem: vscode.StatusBarItem;
+
+export async function setDefaultOrg(): Promise<
+  CancelResponse | ContinueResponse<{}>
+> {
+  let quickPickList = [
+    '$(plus) ' + nls.localize('force_auth_web_login_authorize_org_text'),
+    '$(plus) ' + nls.localize('force_org_create_default_scratch_org_text')
+  ];
+  const defaultDevHubUsernameorAlias = await getDefaultDevHubUsernameorAlias();
+  if (isNullOrUndefined(defaultDevHubUsernameorAlias)) {
+    quickPickList.push(
+      '$(plus) ' + nls.localize('force_auth_web_login_authorize_dev_hub_text')
+    );
+  }
+  const orgList = new OrgList();
+  const authInfoList = await orgList.updateOrgList();
+  if (!isNullOrUndefined(authInfoList)) {
+    quickPickList = quickPickList.concat(authInfoList);
+  }
+
+  const selection = await vscode.window.showQuickPick(quickPickList, {
+    placeHolder: nls.localize('org_select_text')
+  });
+
+  if (!selection) {
+    return { type: 'CANCEL' };
+  }
+  if (
```

Comment:
We could change this to a switch statement but other than that it looks good.

Reply:
Cool, thanks. I can change it to a switch statement and merge it in.